app/controllers/application_controller.rb

```diff
@@ -2,9 +2,156 @@
 
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).

```ruby
class TweetsController < ApplicationController

end
```